src/overlays/actors/ovl_En_Dt/z_en_dt.c

```diff
@@ -28,59 +98,726 @@ ActorInit En_Dt_InitVars = {
+static u16 sTextIds[] = {
+    0x2ABD, 0x2ABE, 0x2ABF, 0x2AC0, 0x2ABE, 0x2AC1, 0x2AC2, 0x2AC3, 0x2AC4, 0x2ABB, 0x2ABC, 0x2AC6, 0x2AC7, 0x2AC8,
+    0x2AC9, 0x2ACA, 0x2ACB, 0x2ACC, 0x2ACD, 0x2ACE, 0x2ACF, 0x2AD0, 0x2AD1, 0x2AC5, 0x2AD3, 0x2AD4, 0x2AD2,
+};
+
+// The code uses EnDt::textIdIndex as an index into this lookup table
+static s32 sCutsceneFocusTargetTable[] = {
+    // Start of initial mayor meeting cutscene
+    /*  0 */ EN_DT_CS_FOCUS_TARGET_BAISEN,
+    /*  1 */ EN_DT_CS_FOCUS_TARGET_MAYOR,
+    /*  2 */ EN_DT_CS_FOCUS_TARGET_MUTO,
+    /*  3 */ EN_DT_CS_FOCUS_TARGET_MUTO,
+    /*  4 */ EN_DT_CS_FOCUS_TARGET_MAYOR,
+    /*  5 */ EN_DT_CS_FOCUS_TARGET_BAISEN,
+    /*  6 */ EN_DT_CS_FOCUS_TARGET_MAYOR,
+    /*  7 */ EN_DT_CS_FOCUS_TARGET_MUTO,
+    /*  8 */ EN_DT_CS_FOCUS_TARGET_MAYOR,
+    // End of intitial mayor meeting cutscene
+
+    // Start of repeat mayor meeting cutscene
+    /*  9 */ EN_DT_CS_FOCUS_TARGET_BAISEN,
+    /* 10 */ EN_DT_CS_FOCUS_TARGET_MUTO,
+    // End of repeat mayor meeting cutscene
+
+    // Start of meeting resolution cutscene
+    /* 11 */ EN_DT_CS_FOCUS_TARGET_BAISEN,
+    /* 12 */ EN_DT_CS_FOCUS_TARGET_MAYOR,
+    /* 13 */ EN_DT_CS_FOCUS_TARGET_MUTO,
+    /* 14 */ EN_DT_CS_FOCUS_TARGET_MAYOR,
+    /* 15 */ EN_DT_CS_FOCUS_TARGET_MUTO,
+    /* 16 */ EN_DT_CS_FOCUS_TARGET_BAISEN,
+    /* 17 */ EN_DT_CS_FOCUS_TARGET_MAYOR,
+    /* 18 */ EN_DT_CS_FOCUS_TARGET_BAISEN,
+    /* 19 */ EN_DT_CS_FOCUS_TARGET_MAYOR,
+    /* 20 */ EN_DT_CS_FOCUS_TARGET_MUTO,
+    /* 21 */ EN_DT_CS_FOCUS_TARGET_MAYOR,
+    /* 22 */ EN_DT_CS_FOCUS_TARGET_MAYOR,
+    // End of meeting resolution cutscene
+
+    // Final night cutscene with mayor
+    /* 23 */ EN_DT_CS_FOCUS_TARGET_MAYOR
+};
+
+static s16 sStringIdCsIndexTable[] = {
+    0x2ABB, 0,  //
+    0x2ABD, 0,  //
+    0x2AC0, 1,  //
+    0x2AC1, 2,  //
+    0x2AC4, 3,  //
+    0x2AC6, 4,  //
+    0x2AC7, 5,  //
+    0x2AC8, 6,  //
+    0x2AC9, 7,  //
+    0x2ACC, 8,  //
+    0x2ACF, 9,  //
+    0x2AD0, 10, //
+};
+
+static s16 sUnused[] = { 777, 777 };
+
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 25, 70, 0, { 0, 0, 0 } },
 };
 
-#endif
+static AnimationHeader* sAnimations[EN_DT_ANIMATION_MAX] = {
+    &object_dt_Anim_00112C, &object_dt_Anim_0005A4, &object_dt_Anim_000854,
+    &object_dt_Anim_000DA8, &object_dt_Anim_000BE0, &object_dt_Anim_00B500,
+};
+
+static u8 sAnimationModes[EN_DT_ANIMATION_MAX] = {
+    ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE,
+};
+
+static s32 sVisualStateTable[] = {
```

Comment:
Might need more opinions, but perhaps something like `sModelInfo`. Not like other ModelInfos I've seen, but it data used to deal with the model.

Reply:
Admittedly I struggled a bit with this one, since I couldn't find another example of this type of state table in any other actor. As far as I can tell, it's unique to this one. That said, I'm not sure I agree `ModelInfo` is an appropriate name for this. If I were to glance at that, I would expect it to contain properties pertaining to the actor's model and/or mesh. Instead, what this actually controls is the actor's look and behavior at key points during the player's interaction with it, including the animation, eye texture, and blinking behavior. 

That said, I'm not committed to the name, and I realize I'm new to the project and don't have a lot of experience with the idioms in it. If people can agree on a better name, I will certainly change it.

Reply:
I took another swing at these names, let me know if they make sense.